OsmAnd/src/net/osmand/plus/views/mapwidgets/widgets/SpeedometerWidget.java

```diff
@@ -455,18 +482,39 @@ private void drawSpeedLimitDescription(Canvas canvas, TextPaint textPaint, Rect
 				unitTextSize = SPEEDOMETER_UNIT_TEXT_SIZE;
 			}
 		}
+		Rect contentRect = getSpeedlimitContentRect(alertRect, newWidgetSize, density);
 		textPaint.setTextSize(unitTextSize * density);
-		float xDescr = alertRect.left + (float) alertRect.width() / 2 - textPaint.measureText(textLimitDescription) / 2;
-		int padding;
-		if (newWidgetSize == WidgetSize.SMALL) {
-			padding = US_SPEED_LIMIT_DESCRIPTION_TOP_S;
-		} else if (newWidgetSize == WidgetSize.MEDIUM) {
-			padding = US_SPEED_LIMIT_DESCRIPTION_TOP_M;
+		Rect textBounds = new Rect();
+		textPaint.getTextBounds(textLimitDescription, 0, textLimitDescription.length(), textBounds);
+		float xDescr = contentRect.left + (float) contentRect.width() / 2 - textPaint.measureText(textLimitDescription) / 2;
+
+		float halfTextHeight = textBounds.height() / 2.0f;
+		float totalWeight = SPEED_LIMIT_DESCRIPTION_WEIGHT + SPEED_LIMIT_VALUE_WEIGHT;
+
+		float calculatedFrameTextHeight = calculateFrameTextHeight(SPEED_LIMIT_DESCRIPTION_WEIGHT, totalWeight, contentRect.height());
+		float frameCenterPosition = (calculatedFrameTextHeight / 2) + halfTextHeight;
+		float yDescr = frameCenterPosition + contentRect.top;
+
+		canvas.drawText(textLimitDescription, xDescr, yDescr, textPaint);
```

Comment:
It looks like some calculations in drawSpeedLimit & drawSpeedLimitDescription are similar, can we try to simplify it?